GitCommands/Git/GitModule.cs

```diff
@@ -3552,64 +3565,52 @@ public bool IsRunningGitProcess()
             return false;
         }
 
-        public static string UnquoteFileName(string fileName)
+        private static readonly Regex _escapedOctalCodePointRegex = new Regex(@"(\\([0-7]{3}))+", RegexOptions.Compiled);
+
+        /// <summary>
+        /// Unescapes any octal code points embedded within <paramref name="s"/>.
+        /// </summary>
+        /// <remarks>
+        /// If no portions of <paramref name="s"/> contain escaped data, then <paramref name="s"/> is returned.
+        /// </remarks>
+        /// <example>
+        /// <code>UnescapeOctalCodePoints(@"\353\221\220\353\213\244") == "두다"</code>
+        /// </example>
+        /// <param name="s">The string to unescape.</param>
+        /// <returns>The unescaped string, or <paramref name="s"/> if no escaped values were present.</returns>
+        [ContractAnnotation("s:null=>null")]
+        public static string UnescapeOctalCodePoints(string s)
         {
-            if (fileName.IsNullOrWhiteSpace())
+            if (s == null)
             {
-                return fileName;
+                return null;
```

Comment:
Why not 
```
if (string.IsNullOrWhiteSpace(s))
{
    return string.Empty;
}
```
?

Reply:
Logic here is that you pass `null`, you get `null`. You pass `""`, you get `""`.

I've been thinking for a while that there's a lot of confusing interchange between empty strings, white-space strings and null strings, both in terms of consumption (i.e. this example) and returns (i.e. catch exception, return `""`). Part of the goal in annotating which arguments can be null is to allow quickly reviewing this across the board. I would aim to tighten restrictions on inputs and outputs over time, and use tooling to identify violations.

Reply:
The rest of the code doesn't look it return `null`.
So to prevent a possible NRE why not return `""`?

Reply:
Changed to return "" though I believe it's contributing to the problem rather than helping. Consistency is good.